docs/_howto/stub-dependency.md

```diff
@@ -52,3 +54,128 @@ describe("moduleUnderTest", function() {
   });
 });
 ```
+
+## A complex example with asynchronous code
+
+In some cases you might need to stub a dependency that returns a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). To test it you can add the [async](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) keyword to the test method and call the method being tested with the [await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await) keyword.
```

Comment:
@mroderick How do you feel about this? I know our codebase is strictly ES5.1, but this is probably how most people write new async code these days, so the examples are more relevant in that regard by using ES2015 syntax.

Reply:
@fatso83 Good point, I didn't even notice I wrote it differently :P It should at least be consistent... Maybe it should be ES5 but very few write code like that...

Reply:
I'm saying ES2015+ is fine in example docs, given that there was no feedback from Morgan.

Whether or not there are few people writing code like that is debatable ... there are very few modern tutorials not emplying a kitchen sink of a build pipeline, but there are definitely looooads of web developers that don't do that, possibly most. But the ones who only make little interactive snippets here and there are not the ones usually employing unit testing in any case :)